src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabHelper.java

```diff
@@ -12,18 +20,45 @@
 public class GitLabHelper {
 
     public static GitLabApi apiBuilder(String serverName) {
+        return apiBuilder(serverName, null);
+    }
+
+    public static GitLabApi apiBuilder(String serverName, StandardCredentials credentials) {
         GitLabServer server = GitLabServers.get().findServer(serverName);
         if (server != null) {
-            PersonalAccessToken credentials = server.getCredentials();
-            if (credentials != null) {
-                return new GitLabApi(server.getServerUrl(), credentials.getToken().getPlainText());
+            credentials = credentials == null ?server.getCredentials(): credentials;
+
+            if (credentials == null) {
+                return new GitLabApi(server.getServerUrl(), GitLabServer.EMPTY_TOKEN);
+            } else if (credentials instanceof PersonalAccessToken) {
+                return new GitLabApi(server.getServerUrl(),((PersonalAccessToken) credentials).getToken().getPlainText());
+            } else if (credentials instanceof UsernamePasswordCredentials) {
+                return new GitLabApi(server.getServerUrl(),((UsernamePasswordCredentials) credentials).getPassword().getPlainText());
```

Comment:
So, this API only takes password? IMO, we can use `PersonalAccessToken` directly.

Reply:
I don't think username password can be used to access GitLab API. If we are to support OAuth2Login we can call this method https://github.com/gitlab4j/gitlab4j-api/blob/master/src/main/java/org/gitlab4j/api/GitLabApi.java#L170

Reply:
Yes, if GitLab API only use PersonalAccessToken, does the credential selection column in the Item only need to list PersonalAccessToken?

Reply:
Yes, I think we should do that but there is a scope for adding oauth support if you are interested.